spec/unit/grpc_web/server/rack_app_spec.rb

```diff
@@ -67,6 +75,18 @@
       end
     end
 
+    context 'given a service module and a lazy init block' do
+      include_context 'given a service module and a lazy init block'
+      it 'validates the class' do
+        expect(::GRPCWeb::ServiceClassValidator).to receive(:validate).with(HelloService::Service)
+        handle
+      end
+
+      it 'raises an error' do
+        expect { app.handle(HelloService) }.to raise_error(ArgumentError, 'HelloService is an abstract interface. You must also provide an initializer block.') # rubocop:disable Layout/LineLength
```

Comment:
This example should go into the `ServiceClassValidator` spec instead if we move the logic there. If not, it should be in a different context, called `'given a service module and no lazy init block'`